runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -6124,24 +6124,33 @@ done:;
 			 */
 			switch (romCPEntry->cpType >> J9DescriptionReturnTypeShift) {
 			case J9DescriptionReturnTypeBoolean:
-				value = (j9object_t)(UDATA)J9VMJAVALANGBOOLEAN_VALUE(_currentThread, value);
+				*(I_32 *)_sp = (I_32)J9VMJAVALANGBOOLEAN_VALUE(_currentThread, value);
```

Comment:
I would prefer to use `(U_32)` to `(I_32)` - no accidental chance of sign extensions.  Though a boolean should be 0 or 1.